libraries/forms.py

```diff
@@ -278,12 +278,30 @@ def _get_top_contributors_for_version(self):
         )
 
     def _get_top_libraries_for_version(self):
-        base_queryset = self.library_queryset.filter(
+        return (
+            self.library_queryset.filter(
+                library_version=LibraryVersion.objects.filter(
+                    library=OuterRef("id"), version=self.cleaned_data["version"]
+                )[:1],
+            )
+            .annotate(commit_count=Count("library_version__commit"))
+            .order_by("-commit_count")
+        )
+
+    def _get_libraries_by_name(self):
+        return self.library_queryset.filter(
             library_version=LibraryVersion.objects.filter(
                 library=OuterRef("id"), version=self.cleaned_data["version"]
             )[:1],
         ).order_by("name")
+
+    def _get_libraries_by_quality(self):
         # returns "great", "good", and "standard" libraries in that order
+        base_queryset = self.library_queryset.filter(
+            library_version=LibraryVersion.objects.filter(
+                library=OuterRef("id"), version=self.cleaned_data["version"]
+            )[:1],
+        )
```

Comment:
There's a lot of duplication here. Could you put the base query into its own method? Something like

```python
    def _get_queryset_with_lib_versions(self):
        return self.library_queryset.filter(
            library_version=LibraryVersion.objects.filter(
                library=OuterRef("id"), version=self.cleaned_data["version"]
            )[:1],
        )
```

Or perhaps `self.library_queryset` should have this filtering on it? Not sure where else it's used.